src/django_components/component.py

```diff
@@ -2478,15 +2862,17 @@ def _render_impl(
         cache_component_css(self.__class__)
         css_input_hash = cache_component_css_vars(self.__class__, css_data) if css_data else None
 
+        #############################################################################
+        # 4. Make Context copy
+        #
+        # NOTE: To support infinite recursion, we make a copy of the context.
+        #       This way we don't have to call the whole component tree in one go recursively,
+        #       but instead can render one component at a time.
+        #############################################################################
+
         with _prepare_template(self, context, template_data, metadata) as template:
             component_ctx.template_name = template.name
 
-            # For users, we expose boolean variables that they may check
-            # to see if given slot was filled, e.g.:
-            # `{% if variable > 8 and component_vars.is_filled.header %}`
-            is_filled = SlotIsFilled(slots_dict)
-            metadata.is_filled = is_filled
```

Comment:
Moved this to [line 2828](https://github.com/django-components/django-components/pull/1205/files#diff-35c84ebceae50f7149f03bbcaf2b4f1289bc2984bf7859ca521b5dbe7960c396R2828)